src/ApiCore/AuthWrapper.php

```diff
@@ -114,6 +118,57 @@ public static function build(array $args = [])
         return new AuthWrapper($credentialsLoader, $authHttpHandler);
     }
 
+    /**
+     * Factory method to create an AuthWrapper from a credentials keyfile and an array of options.
+     *
+     * @param string|array $keyFile
```

Comment:
Hmm, not sure. I wanted to create separate methods to try to make the different construction options clearer, but I agree that the duplicated logic is not good and I'm not sure it really achieves the goal of being clearer. WDYT of trying to unify the naming (e.g. `create()`, `createFromKeyFile()`) and making the options identical in each case? (And also trying to reduce the duplication of logic). Or do you think a single method is just better?

Reply:
I don't want to say a single method is better, it just happens to be my preference here. If we can clean up the logic to reduce duplication, I would be okay with splitting it out. One small note on the naming, though, is that it seems we use `build` for factories elsewhere - maybe it would be helpful to be consistent with that naming here?

Reply:
I have updated to remove as suggested, on reflection it seemed better for the reasons you suggest - PTAL